### PR TITLE
Add skill rate validation

### PR DESCRIPTION
**Pull request by matDobek**
# Ticket
https://netguru.atlassian.net/browse/PEOPLE-126

# Description
Add skill rate validation

### Before merge checklist
- [ ] CircleCI is passing
- [ ] Proper labels are set

### CODE REVIEW

File: app/validators/user_skill_rate/rate_within_range_validator.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+      add_error unless rate_within_range?
+    end
+
+    protected 
+
+    attr_reader :record
+
+    def rate_within_range?
+      rate_type = record.skill&.rate_type
```
Comment (netguru-hound) [inline app/validators/user_skill_rate/rate_within_range_validator.rb]: unexpected token error

File: app/models/skill.rb [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-
   belongs_to :skill_category
   has_many :user_skill_rates
   has_many :users, through: :user_skill_rates
 
   validates :name, :skill_category, :rate_type, presence: true
+  validates :rate_type, inclusion: { in: ::Skills::RateType.stringified_types }
   validate :uniques
 
+
```
Comment (netguru-hound) [inline app/models/skill.rb]: Extra blank line detected.

File: app/validators/user_skill_rate/rate_within_range_validator.rb (hunk 1 above)
Comment (ZuzannaSt) [inline app/validators/user_skill_rate/rate_within_range_validator.rb]: Is this ampersand a typo or some cool feature you could explain?

Comment (matDobek) [inline app/validators/user_skill_rate/rate_within_range_validator.rb]:
```
A safe navigation operator (so-called lonely operator) &., which already exists in C#, Groovy, and Swift, is introduced to ease nil handling as obj&.foo. Array#dig and Hash#dig are also added. Note that this behaves like try! of Active Support, which specially handles only nil.
```

https://www.ruby-lang.org/en/news/2015/12/25/ruby-2-3-0-released/

File: app/models/skills/rate_type.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    TYPES = {
+      'boolean' => 0..1,
+      'range' => 0..4,
+    }.freeze
+
+    def initialize(type:)
+      @type = type.to_s
+    end
+
+    attr_reader :type
```
Comment (siepet) [inline app/models/skills/rate_type.rb]: Maybe move it under `private` if it shouldn't be accessible outside of class?

Comment (matDobek) [inline app/models/skills/rate_type.rb]: actually good point

**Pull request merged**